app/models/project.rb

```diff
@@ -1,4 +1,18 @@
 class Project <ApplicationRecord
   validates_presence_of :name, :material
   belongs_to :challenge
+
+  has_many :contestant_projects
+  has_many :contestants, through: :contestant_projects
+
+  def contestant_experience
+    total_experience = contestants.sum do |contestant|
+      contestant.years_of_experience
+    end.to_f
+    total_experience / contestants.count.to_f
```

Comment:
Try to avoid using Ruby to do calculations. Making calculations is far more efficient if you let the database handle it:

```
contestants.average(:years_of_experience)
```

This code will generate a SQL statement to calculate the average for you. It uses the ActiveRecord#calculate method rather than the Ruby Array#sum method. We will talk more about this when we get to grouping and aggregating.